imgui/src/window/child_window.rs

```diff
@@ -18,12 +18,13 @@ pub struct ChildWindow<'a> {
     border: bool,
 }
 
-impl<'a> ChildWindow<'a> {
```

Comment:
Ah we gotta revert these changes here. For *some* reason, ChildWindows uniquely take `Id` for their `name` instead of `label`. This might be because child windows don't showcase that label anywhere, but is rather bizarre.

How about we leave the bounds on `new` as is (this is how most will consume the API), but let's do `new_with_id` that uses the old `Into<Id>`. Internally, we'll use the `Id` match that we did previously (wrapping the `new` impl AsRef<str> param into an `Id::Str`), and then match on build and go from there?

Reply:
Ah, interesting - BeginChild takes either str or ID,

https://github.com/ocornut/imgui/blob/fc4988ffb048f7d3ebcf9cb96bf7727878ad9d0e/imgui.cpp#L5093-L5103

and both call `BeginChildEx` which explains

https://github.com/ocornut/imgui/blob/fc4988ffb048f7d3ebcf9cb96bf7727878ad9d0e/imgui.cpp#L5061

> If you need to append to a same child from multiple location in the ID stack, use BeginChild(ImGuiID id) with a stable value

..which I understand as meaning `ID` is essentially a global address to a child window, whereas the string option only works relative to the current window.

So.. makes sense, the `new_with_id` thing seems a good approach

Reply:
Uh this makes perfect sense but I have no idea how to do it.

Since ChildWindow would now stores `'&'ui Ui` and an `&'id Id` I end up with something like
```rust
    pub fn child_window<Label: 'ui + AsRef<str>>(&'ui self, name: Label) -> ChildWindow<'ui, '_> {
        ChildWindow::new(self, name)
    }

    pub fn child_window_with_id(&'ui self, id: Id<'id>) -> ChildWindow<'ui, 'id> {
        ChildWindow::new_with_id(self, id)
    }
```

..but in trying to implement these bits on ChildWindow I get baffling errors like

```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'id` due to conflicting requirements
  --> imgui/src/window/child_window.rs:40:43
   |
40 |         let mess: ChildWindow<'ui, 'id> = ChildWindow {
   |                                           ^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime `'id` as defined on the impl at 21:11...
  --> imgui/src/window/child_window.rs:21:11
   |
21 | impl<'ui, 'id> ChildWindow<'ui, 'id> {
   |           ^^^
note: ...so that the expression is assignable
  --> imgui/src/window/child_window.rs:42:13
   |
42 |             id,
   |             ^^
   = note: expected `Id<'_>`
              found `Id<'id>`
note: but, the lifetime must be valid for the lifetime `'ui` as defined on the impl at 21:6...
  --> imgui/src/window/child_window.rs:21:6
   |
21 | impl<'ui, 'id> ChildWindow<'ui, 'id> {
   |      ^^^
note: ...so that the expression is assignable
  --> imgui/src/window/child_window.rs:40:43
   |
40 |           let mess: ChildWindow<'ui, 'id> = ChildWindow {
   |  ___________________________________________^
41 | |             ui,
42 | |             id,
43 | |             flags: WindowFlags::empty(),
...  |
48 | |             border: false,
49 | |         };
   | |_________^
   = note: expected `child_window::ChildWindow<'ui, 'id>`
              found `child_window::ChildWindow<'_, '_>`
```

..which I think is https://github.com/rust-lang/rust/issues/65866 which ends with a not overly reassuring 2 year old comment "Every time I see this message it makes my head explode. So I think my journey with rust will end today" :grimacing:

Reply:
hahahaha holy shit. okay, I'll accept it as it is now, and then I'll schwoop it and try to get more functionality. I can do it at work today